pkg/ccl/crosscluster/logical/dead_letter_queue.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package logical
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdcevent"
+	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+)
+
+// TODO(azhu): Define WriteType as built-in database enum.
+type WriteType int
+
+const (
+	Insert WriteType = iota
+	Delete
+)
+
+func (t WriteType) String() string {
+	switch t {
+	case Insert:
+		return "Insert"
+	case Delete:
+		return "Delete"
+	default:
+		return fmt.Sprintf("Unrecognized WriteType(%d)", int(t))
+	}
+}
+
+type ConflictResolutionType int
+
+const (
+	Unresolved ConflictResolutionType = iota
+	ManuallyResolved
+	Applied
+	Ignored
+)
+
+type DeadLetterQueueClient interface {
+	Create() error
+
+	Log(
+		ctx context.Context,
+		ingestionJobID int64,
```

Comment:
argh, i think we might have miscommunicated. This looks like the api design before we discussed with david and steven. we then decided `dlqClient.Log(ctx, tableID,JobID, kv, cdcEventRow, writeTimestamp, conflictReason)`. See slack message[ here](https://cockroachlabs.slack.com/archives/D0778C89AC8/p1718907044620589?thread_ts=1718906545.570309&cid=D0778C89AC8).

Reply:
with this updated design:
1) no need to grab all that stuff out HandleBatch
2) we will not need to decode a kvBatch twice.
3) we can log `cdcEventRow.DebugString()` for now
4) the call site is still getting many updates but, assume the caller will pass one kv at a time.